Remove HTML style comments from MooseDocs

cticenhour opened the pull request:
These have been deprecated for multiple years. It appears as though patches will have to occur across multiple applications as well as MOOSE itself.

Refs #29573

cticenhour committed:
Remove HTML style comments from MooseDocs comment extension

Change comment implementation method to trigger a RuntimeError when HTML comments are used, until developers are more used to the change.

Refs #29573

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2609586/), step Docs: sync website on 1f0db28 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/29581/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2609633/), step Generate coverage on 1f0db28 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/29581/coverage/xfem)

</details>



This comment will be updated on new commits.

cticenhour committed:
Remove test object documentation from thermal_hydraulics

Many of these were 'old school' stubs, and those which were filled held minor or stock information. None of these are displayable in the documentation, so they're just taking up space.

cticenhour committed:
Remove usage of HTML comments in solid_mechanics

cticenhour committed:
Remove use of HTML comments in modules/doc/content

cticenhour committed:
Remove use of HTML comments in framework, modules/contact, and python docs

cticenhour committed:
Remove use of HTML comments in subchannel module

joshuahansel commented:
Is the new way documented somewhere? I know I've looked for it before and didn't find it, so I just looked around in source files and found the old way.

cticenhour commented:
Good question. I plan to doublecheck the documentation and make sure the markdown-style method is clear as the last step before opening for review! I don't think it's very clear right now, if I'm remembering correctly.

cticenhour committed:
Fixup bulleted list formatting in NearestNodeLocator

cticenhour committed:
Fixup HTML comments after rebase

moosebuild commented:
Job [Documentation internal apps](https://civet.inl.gov/job/3061381/) on ad93ab2 : invalidated by @cticenhour

Pronghorn patched

moosebuild commented:
Job [Documentation controlled apps](https://civet.inl.gov/job/3061380/) on ad93ab2 : invalidated by @cticenhour

RELAP-7 and Sockeye have been patched

cticenhour committed:
Add comment extension documentation and a news entry

cticenhour commented:
Thanks for the reviews!
